test/src/test/java/org/corfudb/runtime/view/AddressSpaceViewTest.java

```diff
@@ -162,4 +164,72 @@ public void readAllWithHoleFill()
         assertThat(m.get(ADDRESS_1).isHole());
         assertThat(m.get(ADDRESS_2).isHole());
     }
+
+    @SuppressWarnings("unchecked")
+    public void holeFillOverwriteIsNotCached()
+            throws Exception {
+        CorfuRuntime r = getRuntime()
+                .setCacheDisabled(false)
+                .connect();
+
+        byte[] testPayload = "hello world".getBytes();
+
+        final long ADDRESS_0 = 0;
+        final long ADDRESS_1 = 1;
+
+        Token token = new Token(ADDRESS_0, r.getLayoutView().getLayout().getEpoch());
+        r.getAddressSpaceView().write(token, testPayload);
+
+        assertThat(r.getAddressSpaceView().read(ADDRESS_0).getPayload(getRuntime()))
+                .isEqualTo("hello world".getBytes());
+
+        Range range = Range.closed(ADDRESS_0, ADDRESS_1);
+        ContiguousSet<Long> addresses = ContiguousSet.create(range, DiscreteDomain.longs());
+
+        Map<Long, ILogData> m = r.getAddressSpaceView().cacheFetch(addresses);
+
+        assertThat(m.get(ADDRESS_0).getPayload(getRuntime()))
+                .isEqualTo("hello world".getBytes());
+
+        Token tokenHoleFilled = new Token(ADDRESS_1, r.getLayoutView().getLayout().getEpoch());
+        assertThatThrownBy(() ->
+                r.getAddressSpaceView().write(tokenHoleFilled, testPayload))
+                .isInstanceOf(OverwriteException.class);
+
+        assertThat(m.get(ADDRESS_1).isHole());
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Complete the assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2970)